server/src/main/java/io/crate/metadata/view/ViewInfo.java

```diff
@@ -94,7 +102,10 @@ public RelationType relationType() {
 
     @Override
     public Iterator<Reference> iterator() {
-        return columns.iterator();
+        return Stream
+            .concat(columns.stream(), subColumns.stream())
+            .sorted(Reference.CMP_BY_POSITION_THEN_NAME)
+            .iterator();
     }
```

Comment:
Didn't touch `columns()` as it returns top columns by the interface contract. 
Pretty much same with `DocTableInfo`